### ISSUE TITLE
Crimson Scale Scenario 5

### ISSUE THREAD

**Issue by JP-Ellis** [Lurkars/gloomhavensecretariat#168]
Here's this week's update for scenario 5:

- [x] The Blood Ooze is meant to be a boss and have:
  - [x] HxC hit point where H is an elite b lo ood ooze
  - [x] Draws from the boss deck
  - [x] Immune to all negative conditions
  - [ ] Immune to all damage (if that's even possible in Gloomhaven Secratariat?)
- [ ] Flaming drake has a 🛡️+2+2 which I'm pretty sure is an error

**Comment by Lurkars** [comment]
Will check for the special ooze. Can you explain more on the flaming Drake shield issue? Don't understand and cannot reproduce. No shield on stats and only one ability with fixed shield 2.

**Comment by Lurkars** [comment]
@JP-Ellis do you mean Scenario 5 maybe?

**Comment by JP-Ellis** [comment]
Yes 🤦‍♂️ my bad, and fixed.

**Comment by Lurkars** [comment]
Okay, this is the problem with the parsed standee data. I have all the special bosses etc. But their not setup. I will go through all data now and look for this. Anyway, the flaming drake questions remain!

**Comment by JP-Ellis** [comment]
Thanks for all the amazing work!

With this scenario in particular, there is one additional feature that would be awesome (though I have no idea if easily implemented).

<details>
<summary>Spoiler</summary>

The boss ooze is spawned into the room initially and is untouchable (no negative conditions, no damage allowed) until specific conditions have been met.

After these conditions are met, the boss is replaced by an 'super' elite ooze with the same HP as the boss and starts drawing from the blood ooze deck.

If it is at all possible, it would be amazing to have one of the section buttons at the bottom do this swap automatically. Of course this can be done manually if that is too difficult or niche to implement.

</details>

**Comment by Lurkars** [comment]
Definitely possible with current features, I will add this. 

Anyway, I still need more information on the mentioned gaming flaming drake issue!

**Comment by JP-Ellis** [comment]
Thanks!

I was trying to replicate the issue with the flaming drake. I believe it was with card 1209, but my attempts to replicate this have failed so far. So not sure whether it was a temporary glitch, or the issue was resolved independently.

Happy for you to close this issue :)

**Comment by Lurkars** [comment]
Hey thanks.

Have a few question for the #5 special rules:
- the immunity to all conditions only last until damage also allowed right? Currently the giant ooze is always immune to all, not sure how to interpret the word `and` correct in the text.
- `At the end of every other round, spawn one elite Blood Ooze` also a bit unsure how "end of the other round" is meant. I understand that I open the section in Round 1, then I play Round 1, next Round is 2 and at the end of this I spawn the first Ooze. So I am playing 2 rounds without spawning (opening round 1 until end, then round 2 until end, then spawn and have at start of Round 3 the first ooze!)

**Comment by Lurkars** [comment]
I implemented now in [0.52.2](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.52.2) as stated above, so on opening room, Giant is not affect-able by nothing, but already acting. After end of second round, spawning of elite B-O begins. When C of those are killed, the Giant become affect-able by everything.

The increase of Max HP function is not implemented yet, this will take some more steps. So still referring to Book there and has to be done manually. Anyway, please check if this is first: correct interpreted and second correct implemented.

**Comment by JP-Ellis** [comment]
Thanks, much appreciate the work you're putting in this.

As a first observation, I personally don't expect Gloomhaven Secretariat to do everything automatically. I think it is perfectly reasonable that some special rules are left to the players to implement themselves. This in itself I think is a feature because if the logic is too restrictive, we can't play special fan-made scenarios shared online.

Having said that, I've tested out the logic for this scenario this morning and there are a couple of errors I think:

- [ ] The gelatinous giant has no HP marker anymore, and no ability to increase its HP. This is problematic because one of the special attacks is that it damages all other blood oozes and increases its _max HP_ equal to the damage inflicted. With no ability to increase max HP itself, it makes it difficult to track how much HP it has gained for when it eventually turns into an overpowered elite blood ooze.

- [ ] The elite blood oozes are spawned "every alternate round". That is in itself a bit ambiguous (my party understood it to mean the next round after entering the room, and every 2nd round thereafter), but the prompt to spawn the elite oozes appears every round. This in itself isn't necessarily an issue as it leaves the interpretation of the rules up to the players, but there's also no ability to dismiss a single appearance of the prompt within dismissing them all. I can see two alternate fixes to this:

  - [ ] Adding a 'Dismiss' button be added next to 'Apply'. This would be particularly useful if the player already applied the action not realising that Gloomhaven Secretariat can do it automatically)
  
    ![image](https://user-images.githubusercontent.com/3196162/226743090-4d1f7959-9a7e-48a2-b2e7-6a386b1fd346.png)

  - [ ] Changing the logic so that if players open the door on round $R$, the prompt only appears on rounds $R+1$, $R+3$, $R+5$, etc.

- [ ] The conversion of the gelatinous giant to the elite ooze did not go as planned. The rules save that the gelatinous giant is converted into an (elite) blood ooze with the same max HP as the boss. A sequence of events in this scenario would be:
  - Gelatinous giant is spawned in at $H \times C$ max HP
  - Through its special ability, it gains an additiona $n$ max HP (this the players can do manually). It now has $H \times C + n$ max HP
  - Once the $C$ elite blood oozes are killed, the prompt appears which will replace the gelatinous giant by an elite blood ooze with $H \times C + n$ max HP.

**Comment by Lurkars** [comment]
Couldn't let this go and tried to implement most of the stuff. I still have one issue, where I don't know how to solve now. I thought, on C players, C elite Blood Oozes must be killed for the Giant Ooze to transform. But since the Oozes spawns in the closest `a` tiles, you can have 4 oozes in the first water section above the Giant Ooze, kill them all 4 will only count as 1, right? No idea how to track those... I could like create 4 spawn dialogs where you chose a specific `a` tile or so. 

> * Adding a 'Dismiss' button be added next to 'Apply'. This would be particularly useful if the player already applied the action not realising that Gloomhaven Secretariat can do it automatically)

You can dismiss every rule just by clicking on the text of it :sweat_smile: (On desktop came up a corresponding cursor)

**Comment by JP-Ellis** [comment]
Oh, I simply didn't try to click outside of the prompt... but that makes sense.

Regarding the rules, we have a different interpretation. I'm pretty sure just _one_ Elite Ooze is spawned every 2nd round, and it is spawned in the closest 'a' tile to the Gelatinous Giant (or the nearest water tile to said 'a' tile, should that tile not be available as a spawn). I don't think that $C$ Elite Oozes spawn, as that would make the scenarion _extremely_ difficult.

So to summarise, here's how I see some rounds going:

| Round | Events |
| ----- | ------ |
| $R$     | Door is opened <br/> Gelatinous Giant is invincible |
| $R + 1$ | 1 Elite Blood Ooze spawns |
| $R + 2$ | No special actions |
| $R + 3$ | 1 Elite Blood Oooze spawns |
| ... | ... |
| $R + n$ | $C$ Elite Blood Oozes are killed <br/> Gelatinous Giant is replaced by an Elite Blood Oooze with HP equal to Gelatinous Giant |

**Comment by Lurkars** [comment]
Yes, only 1 is spawned, but rules say: in the closest `a` water tile hexes. So looking at the map, the one on the top has 4 water tiles, it's closest to giant ooze's starting position. Just take theoretically: giant ooze does not move and you don't kill any elite ooze, the next spawns would all spawn on that top water tiles, aren't they? And you have to destroy all 4 `a`s by killing there spawned oozes. Yes that's my interpretation, if you say, you just have to kill *C* of the spawned oozes, I am happy because this is currently implemented!

**Comment by JP-Ellis** [comment]
I don't think the placement matters too much for Gloomhaven Secrariat, but here's a breakdown of my party's understanding (though I don't have the rules with me at the moment to verify):

1. Select the closest `a` tile to the Gelatinous Giant
   1. If two are equidistant, take the closest `a` tile to the focus. If two are still equidistant, then it is up to the players
2. Identify the closest water tile to the `a` tile identified above (within the same patch of water) in which a spawn is allowed. This will generally be the `a` tile itself, but if it is already occupied, it would be typically be one of the adjacent tiles.

Each time an Elite Blood Oooze is killed, a patch of water is removed (up to the players to decide on this).

Hope that helps 😄
